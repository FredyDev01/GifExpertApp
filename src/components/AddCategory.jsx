import PropTypes from 'prop-types'
import { useState } from 'react'


export const AddCategory = ({onNewCategory})=> {
    
    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault() 
        if(inputValue.trim().length == 0) return
        onNewCategory(inputValue.trim())
        setInputValue('')
    }
    
    return(
        <form onSubmit={onSubmit} aria-label="form">
            <input type="search" placeholder="Buscar gifs :D" value={inputValue} onChange={(event) => onInputChange(event)}/>
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}