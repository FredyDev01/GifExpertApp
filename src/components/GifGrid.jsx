import { GifItem } from "./"
import { useFetchGifs } from "../hooks/useFetchGifs"


export const GifGrid = ({categ})=> {
    const {images, isLoading} = useFetchGifs(categ)   

    return(        
        <>
            <h3>{categ} <hr/></h3>
            {
                isLoading && ( <h2>Cargando</h2> )
            }
            <div className="card-grid">                                
                {
                    images.length ?
                    images.map(image => (
                        <GifItem key={image.id} {...image}/>
                    ))
                    :
                    <p className="infoData">La busqueda ingrezada no tiene ningun dato registrado.</p>                    
                }                
            </div>
        </>                
    )
}