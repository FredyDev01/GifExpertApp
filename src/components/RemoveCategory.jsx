export const RemoveCategory = ({onDeleteCategory})=> {
    const onClickBtn = (val)=> {
        onDeleteCategory(val)
    }

    return(
        <div className="ContentOptions">
            <button className="BtnDel" onClickCapture={()=> {onClickBtn(false)}}>Borrar busqueda</button>
            <button className="BtnDelAll" onClickCapture={()=> {onClickBtn(true)}}>Borrar todo</button>
        </div>
    )
}