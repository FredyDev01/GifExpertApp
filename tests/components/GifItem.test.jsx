import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Pruebas en <GifItem />', () => {
    const title = 'Gato bailarin'
    const url = 'https://media.tenor.com/5Jingks2rHAAAAAM/%D1%84%D0%BB%D0%B5%D0%BA%D1%81%D0%BA%D0%BE%D1%82%D0%B0-catflex.gif'

    test('debe de hacer match en el snapshot', ()=> {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('debe de mostrar la imagen con URL indicado y el ALT indicado', ()=> {
        render(<GifItem title={title} url={url} />)
        console.log('url de la imagen', screen.findByRole('img').src)
        //expect(screen.getByRole('img').src).toBe(url)
        //expect(screen.getByRole('img').alt).toBe(title)
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('debe de haber un parrafo con el title', ()=>{
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()
    })
})