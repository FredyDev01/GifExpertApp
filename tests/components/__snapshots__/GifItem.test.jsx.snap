// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match en el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gato bailarin"
      src="https://media.tenor.com/5Jingks2rHAAAAAM/%D1%84%D0%BB%D0%B5%D0%BA%D1%81%D0%BA%D0%BE%D1%82%D0%B0-catflex.gif"
    />
    <p>
      Gato bailarin
    </p>
  </div>
</div>
`;
